### PR TITLE
Revert "core/assert: remove panic include from assert.h"

### LINKED ISSUE #18175: assert() always prints location of core/lib/assert.c:29
#### Description
A failing assertion should print the memory location where the `assert()` was called.
Instead it always prints the memory location of where `core_panic()` was called inside `assert()`

#### Steps to reproduce the issue
Trigger any assertion.

#### Expected results
The address printed resolves to the location of the failed `assert()`  statement (e.g. with `addr2line`).

#### Actual results
The address printed resolves to `core/lib/assert.c:29`.

### PR DESCRIPTION
**Pull request by benpicco**
### Contribution description

This reverts commit 0b52b3e62e93075bc6c8fba369cde8636c04e524 from #17574 which broke the `assert()` memory location printing.

### Testing procedure

Add an `assert(0)` to e.g. `examples/hello-world`

```patch
--- a/examples/hello-world/main.c
+++ b/examples/hello-world/main.c
@@ -20,11 +20,14 @@
  */
 
 #include <stdio.h>
+#include <assert.h>
 
 int main(void)
 {
     puts("Hello World!");
 
+    assert(0);
+
     printf("You are running RIOT on a(n) %s board.\n", RIOT_BOARD);
     printf("This board features a(n) %s MCU.\n", RIOT_MCU);
 
```

On `master` this prints

```
2022-06-10 22:57:31,223 # main(): This is RIOT! (Version: 2022.04-devel-145-g0b52b-HEAD)
2022-06-10 22:57:31,224 # Hello World!
2022-06-10 22:57:31,224 # 0x2af
2022-06-10 22:57:31,226 # *** RIOT kernel panic:
2022-06-10 22:57:31,228 # FAILED ASSERTION.
2022-06-10 22:57:31,228 # 
2022-06-10 22:57:31,229 # *** halted.
```

    % arm-none-eabi-addr2line -e bin/same54-xpro/*.elf 0x2af
    /home/benpicco/dev/RIOT/core/assert.c:29

With this reverted it prints the location of the `assert()` again:

```
2022-06-10 23:01:03,009 # main(): This is RIOT! (Version: 2022.07-devel-748-g4c5bd)
2022-06-10 23:01:03,010 # Hello World!
2022-06-10 23:01:03,010 # 0x275
2022-06-10 23:01:03,012 # *** RIOT kernel panic:
2022-06-10 23:01:03,014 # FAILED ASSERTION.
2022-06-10 23:01:03,014 # 
2022-06-10 23:01:03,015 # *** halted.
```

    % arm-none-eabi-addr2line -e bin/same54-xpro/*.elf 0x275
    /home/benpicco/dev/RIOT/examples/hello-world/main.c:29

### Issues/PRs references
fixes #18175

### CODE REVIEW

**Review by maribu — APPROVED** [review]

**Comment by benpicco** [comment]
Huh why is `examples/gcoap` now almost 1k larger with this patch?

**Comment by benpicco** [comment]
Turns out the size decrease was introduced by https://github.com/RIOT-OS/RIOT/commit/0b52b3e62e93075bc6c8fba369cde8636c04e524 too - and that makes sense, as it made all `asserts()` print the same output.

File: core/lib/include/assert.h [hunk 1]
```diff
@@ -22,6 +22,8 @@
 #ifndef ASSERT_H
 #define ASSERT_H
 
+#include "panic.h"
```
Comment (kfessel) [inline core/lib/include/assert.h:25]: please do not in include any other header in libc lookalike header libc (by default)

Comment (kfessel) [reply core/lib/include/assert.h:25]: http://www.open-std.org/jtc1/sc22/wg14/www/C99RationaleV5.10.pdf :7.1.3 p 112 (pdf:119) "...the Standard assures that macro and typedef names are reserved only if the associated header is explicitly included"

"Also reserved for the implementor are all external identifiers beginning with an underscore, and all other identifiers beginning with an underscore followed by a capital letter or an underscore." (same doc same §)

these buildins kind of things of gcc are __ : __attribute__ __LINE__
glibc assert.h uses __assert_fail

see dissicusion in  #17574

**Review by kfessel — COMMENTED** [review]
i suggest fixing this another way

**Comment by kfessel** [comment]
did you enable verbose assert? (when you received all messages the same)

**Review by kfessel — CHANGES_REQUESTED** [review]
please copy the line the generate the message you like to see from `panic.c`:65  to `assert.c` 
```
            cpu_print_last_instruction();
```
seems to be the one

**Pull request closed without merging**